src/ODETools/ODESolutions.jl

```diff
@@ -70,3 +70,55 @@ function Base.iterate(sol::GenericODESolution, state)
 
   return (uf, tf), state
 end
+
+# Specialization for a 2nd order ODE
+
+struct Generic2ndOrderODESolution <: ODESolution
+  solver::ODESolver
+  op::SecondOrderODEOperator
+  u0::AbstractVector
+  v0::AbstractVector
+  a0::AbstractVector
+  t0::Real
+  tF::Real
+end
```

Comment:
As above, GenericODESolution could have the order N as template parameter and a tuple of N abstract vectors. This way, I think you don't need to replicate the code for N=1 and N=2. On the other side, I don't think we want to go more than second order, so it is probably too much.

Reply:
Done in the new approach